feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -426,228 +443,234 @@ func TestTempSensor(t *testing.T) {
+				} else {
+					id := card.GetId()
+					t.Logf("Component %s Id: %s", cName, id)
+					if id == "" {
+						t.Errorf("Component %s Id: got empty string, want non-empty string", cName)
+					}
 				}
 			}
-		}
 
-		if p.nameValidation {
-			name := card.GetName()
-			t.Logf("Component %s Name: %s", cName, name)
-			if name == "" {
-				t.Errorf("Encountered empty Name for component %s", cName)
+			if p.nameValidation {
+				name := card.GetName()
+				t.Logf("Component %s Name: %s", cName, name)
+				if name == "" {
+					t.Errorf("Encountered empty Name for component %s", cName)
+				}
 			}
-		}
 
-		if p.partNoValidation {
-			partNo := card.PartNo
-			t.Logf("Component %s PartNo: %v", cName, partNo)
-			if partNo == nil {
-				if card.GetType() == fanType {
-					fanTray := card.GetParent()
-					fanTrayPartNo := gnmi.Lookup(t, dut, gnmi.OC().Component(fanTray).PartNo().State())
-					fanTrayPartNoVal, ftpnPresent := fanTrayPartNo.Val()
-					if !ftpnPresent {
-						t.Errorf("PartNo for Component %s and its parent: got empty string, want non-empty string", cName)
+			if p.partNoValidation {
+				partNo := card.PartNo
+				t.Logf("Component %s PartNo: %v", cName, partNo)
+				if partNo == nil {
+					switch card.GetType() {
+					case componentType["Fan"]:
+						fallthrough
+					case componentType["Storage"]:
+						fallthrough
+					case componentType["Cpu"]:
+						parent := card.GetParent()
+						ppn, ok := gnmi.Lookup(t, dut, gnmi.OC().Component(parent).PartNo().State()).Val()
+						t.Logf("Component %s (parent of %s) PartNo: %v", parent, cName, ppn)
+						if !ok {
+							t.Errorf("PartNo for Component %s and its parent: got empty string, want non-empty string", cName)
+						}
+					default:
+						t.Errorf("PartNo for Component %s: got empty string, want non-empty string", cName)
 					}
-					t.Logf("Component %s (parent of %s) PartNo: %v", fanTray, cName, fanTrayPartNoVal)
-				} else {
-					t.Errorf("PartNo for Component %s: got empty string, want non-empty string", cName)
 				}
 			}
-		}
 
-		if p.serialNoValidation {
-			serialNo := card.SerialNo
-			t.Logf("Component %s SerialNo: %v", cName, serialNo)
-			if serialNo == nil {
-				if card.GetType() == fanType {
-					fanTray := card.GetParent()
-					fanTraySerialNo := gnmi.Lookup(t, dut, gnmi.OC().Component(fanTray).SerialNo().State())
-					fanTraySerialNoVal, ftsnPresent := fanTraySerialNo.Val()
-					if !ftsnPresent {
-						t.Errorf("SerialNo for Component %s and its parent: got empty string, want non-empty string", cName)
+			if p.serialNoValidation {
+				serialNo := card.SerialNo
+				t.Logf("Component %s SerialNo: %v", cName, serialNo)
+				if serialNo == nil {
+					switch card.GetType() {
+					case componentType["Fan"]:
+						fallthrough
+					case componentType["Storage"]:
+						fallthrough
+					case componentType["Cpu"]:
+						parent := card.GetParent()
+						psn, ok := gnmi.Lookup(t, dut, gnmi.OC().Component(parent).SerialNo().State()).Val()
```

Comment:
Same comment as above. Also, unless I'm missing something, the PR also seems to be expecting that the immediate parent does contain a serial number/part number, no? What if the parent is a motherboard that is not included in the inventory? The same solution should suffice here where you may end up searching until you find the part number (i.e. maybe to the linecard, without an assumption as to the specific vendor hierarchy)

Reply:
We should stop at first parent which is removable for serial number also.

Reply:
Done!